File: TypeScript/life.js
var Kid = /** @class */ (function () {
    function Kid(name, age) {
        this.name = name;
        this.age = age;
    }
    Kid.prototype.getInfo = function () {
        return "".concat(this.name, " (").concat(this.age, " years old)");
    };
    return Kid;
}());
var Wife = /** @class */ (function () {
    function Wife(name, age) {
        this.name = name;
        this.age = age;
        this.kids = [];
    }
    Wife.prototype.addChild = function (name, age) {
        var kid = new Kid(name, age);
        this.kids.push(kid);
    };
    Wife.prototype.getKidsInfo = function () {
        return this.kids.map(function (kid) { return kid.getInfo(); }).join(", ");
    };
    Wife.prototype.getInfo = function () {
        return "".concat(this.name, " (Wife, ").concat(this.age, " years old)");
    };
    return Wife;
}());
var Husband = /** @class */ (function () {
    function Husband(name, age) {
        this.name = name;
        this.age = age;
        this.wife = null;
    }
    Husband.prototype.marry = function (wife) {
        this.wife = wife;
    };
    Husband.prototype.getInfo = function () {
        if (this.wife) {
            return "".concat(this.name, " (Husband, ").concat(this.age, " years old) - Married to ").concat(this.wife.getInfo());
        }
        else {
            return "".concat(this.name, " (Husband, ").concat(this.age, " years old) - Single");
        }
    };
    return Husband;
}());
var wife = new Wife("Sneha", 27);
var husband = new Husband("Madhav", 30);
husband.marry(wife);
wife.addChild("John", 5);
wife.addChild("Clair", 3);
console.log(husband.getInfo());
console.log("Children: ".concat(wife.getKidsInfo()));

File: Python/life.py
class Kid:
    def __init__(self, name: str, age: int):
        self.name = name
        self.age = age

    def get_info(self) -> str:
        return f"{self.name} ({self.age} years old)"


class Wife:
    def __init__(self, name: str, age: int):
        self.name = name
        self.age = age
        self.kids = []

    def add_child(self, name: str, age: int) -> None:
        kid = Kid(name, age)
        self.kids.append(kid)

    def get_kids_info(self) -> str:
        return ", ".join(kid.get_info() for kid in self.kids)

    def get_info(self) -> str:
        return f"{self.name} (Wife, {self.age} years old)"


class Husband:
    def __init__(self, name: str, age: int):
        self.name = name
        self.age = age
        self.wife = None

    def marry(self, wife: Wife) -> None:
        self.wife = wife

    def get_info(self) -> str:
        if self.wife:
            return f"{self.name} (Husband, {self.age} years old) - Married to {self.wife.get_info()}"
        else:
            return f"{self.name} (Husband, {self.age} years old) - Single"


wife = Wife("Sneha", 27)
husband = Husband("Madhav", 30)

husband.marry(wife)
wife.add_child("John", 5)
wife.add_child("Clair", 3)

print(husband.get_info())
print(f"Children: {wife.get_kids_info()}")
